### PR TITLE
Replace circleci with github actions

### PR DESCRIPTION
**Pull request by Stoops-ML**
For code coverage to work a `CODECOV_TOKEN` needs to be added to secrets.

A commit with a `v1.0.0` tag should be added to the repo to trigger the GitHub action to publish to PyPi.

### CODE REVIEW

**Comment by astrojuanlu** [comment]
YES - could you send a dummy PR right after we merge this one just to test if all the checks pass?

**Comment by astrojuanlu** [comment]
Also please remove all the CircleCI config

**Comment by Stoops-ML** [comment]
> YES - could you send a dummy PR right after we merge this one just to test if all the checks pass?

Yep

**Review by astrojuanlu — APPROVED** [review]

**Pull request merged**
